add support for Loki ruler configuration

weakcamel opened the pull request:
update also broken Loki documentation link

weakcamel committed:
add support for Loki ruler configuration

update also broken Loki documentation link

weakcamel commented:
@diogenxs Not sure if you care about this role anymore, but I thought I'd try to merge my changes back, just in case.

P.S. Thank you for writing it in the first place, it's very handy!

weakcamel committed:
download loki binaries on the target system

this has the downside of taking up a bit of space,
but at least this makes the role more idempotent when it comes
to re-running it from different control nodes (e.g. in CI)

weakcamel committed:
update download task name

weakcamel committed:
update README with author information

weakcamel committed:
update to Github Actions to release

weakcamel committed:
Merge pull request #1 from weakcamel/idempotent_loki_install

weakcamel committed:
update role metadata

weakcamel committed:
fix unarchiving to run on host and not controller

rutger1140 commented:
@weakcamel Nice work on the updates. I'm actually only looking for adding the `ruler` setting. I'll create another PR for that.

I can imagine that @diogenxs won't approve your PR when you remove his name from the `meta/main.yml` file 😛 😉

weakcamel commented:
> I can imagine that @diogenxs won't approve your PR when you remove his name from the `meta/main.yml` file 😛 😉

Lol, that's  a good point. I actually assumed that the original author is no longer interested in maintaining this role so I need to go with my fork anyway. Should I be wrong though, i'm happy to revert that bit in this PR (or someone else to revert that).

weakcamel committed:
add query_range, query_scheduler and custom configs

query_range and query_scheduler for the sake of
https://github.com/grafana/loki/issues/4613

'custom_config' is for all future sections which may be added
and aren't available at the moment of publishing this role

weakcamel committed:
fix Github actions - part1

- update lint action's Github branch to main
- fix FQCN findings

weakcamel committed:
add required task names to blocks

weakcamel committed:
install: fix file permissions (lint finding)

weakcamel committed:
Merge pull request #3 from weakcamel/update_github_actions_and_fix_lint

fix Github actions

weakcamel committed:
bump minimum Ansible version (due to FQCN)

weakcamel committed:
Merge pull request #2 from weakcamel/add_extra_loki_config

add query_range, query_scheduler and custom configs

weakcamel committed:
bump galaxy-action version

weakcamel committed:
replace include with import_tasks

weakcamel committed:
update meta (namespace name)

weakcamel committed:
Merge pull request #4 from weakcamel/replace_includes

replace include with import_tasks

weakcamel committed:
address all ansible-lint findings

weakcamel committed:
bump github action versions

weakcamel committed:
Merge pull request #5 from weakcamel/fix_linter_errors

address all ansible-lint findings

Commit:
fix: added missing dependency

weakcamel committed:
Merge pull request #6 from anon8675309/missing-unzip

fix: added missing dependency

weakcamel committed:
add compactor settings to loki config

weakcamel committed:
Merge pull request #8 from weakcamel/add_compactor_settings

add compactor settings to loki config

weakcamel committed:
Deprecation of this role - update readme
